px-map-behavior-layer-geojson.es6.js

```diff
@@ -149,29 +198,46 @@
 
         style: (feature) => {
           const featureProperties = feature.properties.style || {};
+          const attributeProperties = this.getInstOptions().featureStyle;
 
-          return this._getStyle(featureProperties, styleAttributeProperties);
+          return this._getStyle(featureProperties, attributeProperties);
         }
       });
+
       if(this.editable) {
-        if (!this.parentNode.elementInst.editTools) {
-          this.parentNode.elementInst.editTools = new L.Editable(this.parentNode.elementInst, {featuresLayer: geojsonLayer});
-        } else {
-          this.parentNode.elementInst.editTools.featuresLayer.addLayer(geojsonLayer);
+        if(!this.parentNode.elementInst.editTools) {
+          if(this.sketch) {
+            this.parentNode.elementInst.editTools = new L.Editable(this.parentNode.elementInst, {featuresLayer: geojsonLayer});
+          } else {
+            this.parentNode.elementInst.editTools = new L.Editable(this.parentNode.elementInst);
+          }
+          //Disable doubleclick zoom when drawing to prevent zooming when double clicking to end a line
+          this.parentNode.elementInst.editTools.addEventListener('editable:drawing:start', () => {
+            this.parentNode.elementInst.doubleClickZoom.disable();
+          });
+
+          this.parentNode.elementInst.editTools.addEventListener('editable:drawing:end', () => {
+            //0ms timeout to ensure that double clicking doesn't zoom when placing vertex and immeditaley finishing drawing
+            setTimeout(() => {
+              this.parentNode.elementInst.doubleClickZoom.enable();
+            },0);
+          });
+
+          this.parentNode.elementInst.editTools.addEventListener('editable:dragstart', () => {
+            this.parentNode.elementInst.doubleClickZoom.disable();
+          });
+
+          this.parentNode.elementInst.editTools.addEventListener('editable:dragend', () => {
+            //0ms timeout to ensure that double clicking doesn't zoom when placing vertex and immeditaley finishing drawing
+            setTimeout(() => {
+              this.parentNode.elementInst.doubleClickZoom.enable();
+            },0);
+          });
+        } else if(this.sketch) {
+          this.parentNode.elementInst.editTools.featuresLayer = geojsonLayer;
         }
-        //Disable doubleclick zoom when drawing to prevent zooming when double clicking to end a line
-        this.parentNode.elementInst.editTools.addEventListener('editable:drawing:start', () => {
-          this.parentNode.elementInst.doubleClickZoom.disable();
-        });
-
-        this.parentNode.elementInst.editTools.addEventListener('editable:drawing:end', () => {
-          //0ms timeout to ensure that double clicking doesn't zoom when placing vertex and immeditaley finishing drawing
-          setTimeout(() => {
-            this.parentNode.elementInst.doubleClickZoom.enable();
-          },0);
-        });
-
       }
+
```

Comment:
A general note that this function is now quite long, can we move the Editable checking into its own function and call it from here?